toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -276,16 +324,38 @@ func (im *IsoMaker) prepareWorkDirectory() {
 // copyStaticIsoRootFiles copies architecture-independent files from the
 // Mariner repo directories.
 func (im *IsoMaker) copyStaticIsoRootFiles() {
-	staticIsoRootFilesPath := filepath.Join(im.resourcesDirPath, "assets/isomaker/iso_root_static_files/*")
 
-	logger.Log.Debugf("Copying static ISO root files from '%s'.", staticIsoRootFilesPath)
+	if im.resourcesDirPath == "" && im.grubCfgPath == "" {
+		logger.Log.Panicf("Missing required parameters. Must specify either the resources directory or provide a grub.cfg.")
```

Comment:
If we are going to make this code into a proper module/library, then all of these panics really ought to be removed and replaced with proper error handling.

Reply:
Makes sense... will change.

Reply:
Created a draft PR here for the elimination of 'panic': https://github.com/microsoft/CBL-Mariner/pull/7454/files
Of special interest is the `defer` function error propagation (since the original intent was to stop execution when a clean-up fails).
@cwize1